tests/validation/tests/v3_api/common.py

```diff
@@ -2084,15 +2084,28 @@ def readDataFile(data_dir, name):
         return f.read()
 
 
-def set_url_password_token(rancher_url, server_url=None):
+def set_url_password_token(rancher_url, server_url=None, version="oldUI"):
     """Returns a ManagementContext for the default global admin user."""
     auth_url = \
         rancher_url + "/v3-public/localproviders/local?action=login"
-    r = requests.post(auth_url, json={
-        'username': 'admin',
-        'password': 'admin',
-        'responseType': 'json',
-    }, verify=False)
+    rpassword = 'admin'
+    print(auth_url)
+    if version.find("master") > -1 or version.find("2.6") > -1:
+        rpassword = ADMIN_PASSWORD
+        print("on 2.6 or later")
+    retries = 4
+    for attempt in range(retries):
```

Comment:
`password request failed. Retry attempt:  0 of 4`
Nit but it would be more readable if we use `range(1, retries)` with `retries = 5`

We have a `wait_for` in `tests/common` wouldn't that be better to use it?

Reply:
from what I saw, the `wait_for` function wouldn't handle the error being thrown, it was only for waiting for the callback to be available. I did update the range in the latest version

Reply:
You can handle the error in the callback and return `False/None` or a `True` depending on the request result.